src/server/project.ts

```diff
@@ -292,6 +306,14 @@ namespace ts.server {
             return this.projectService.typingsCache;
         }
 
+        /*@internal*/
+        getProbableSymlinks(files: readonly SourceFile[]): ReadonlyMap<string> {
+            return this.symlinks || (this.symlinks = moduleSpecifiers.discoverProbableSymlinks(

```

Comment:
Does PackageJson cache code need to be in compiler next to moduleSpecifiers instead as it may be needed for .d.ts emit?

Reply:
Auto-imports (changes added in this PR) is the only time `discoverProbableSymlinks` was getting called in a big loop—one time for every package in node_modules, all in an interactive scenario. It could probably be cached on Program instead and might marginally speed up some emit in some scenarios. Is that what you’re suggesting?

Reply:
Sorry missed seeing this. Yeah What I meant was symLinks and/or packageJson could probably cached on program when `host` doesn't have `getProbableSymlinks` and passed that during declaration emit?

Reply:
Makes sense. Would you start with a fresh cache on every program, or do you think it would be safe to recycle the old cache based on `oldProgram.structureIsReused`?

Reply:
Yes on every program since the projects without proper watches cant handle it

Reply:
Done 👍